addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,193 @@
+                                            "/." + PATH + "/./",
+                                            PATH + "%20/",
+                                            "/%20" + PATH + "%20/",
+                                            PATH + "..;",
+                                            PATH + "?",
+                                            PATH + "??",
+                                            "/" + PATH + "//",
+                                            PATH + "/",
+                                            PATH + "/.testus",
+                                            PATH + "../app.py"
+                                        }));
+                List<String> hpayloads =
+                        new LinkedList<String>(
+                                Arrays.asList(
+                                        new String[] {
+                                            "X-Rewrite-URL: " + PATH,
+                                            "X-Original-URL: " + PATH,
+                                            "Referer: " + PATH,
+                                            "X-Custom-IP-Authorization: 127.0.0.1",
+                                            "X-Originating-IP: 127.0.0.1",
+                                            "X-Forwarded-For: 127.0.0.1",
+                                            "X-Remote-IP: 127.0.0.1",
+                                            "X-Client-IP: 127.0.0.1",
+                                            "X-Host: 127.0.0.1",
+                                            "X-Forwared-Host: 127.0.0.1"
+                                        }));
+
+                // sending payloads in URL.
+                for (String i : upayloads) {
+                    String TMP_URI = SCHEMA + "://" + HOST + "" + i;
+                    HttpMessage req_with_payload =
+                            new HttpMessage(new org.apache.commons.httpclient.URI(TMP_URI, true));
+                    sendAndReceive(req_with_payload);
+                    if (req_with_payload.getResponseHeader().getStatusCode() == HttpStatusCode.OK) {
+                        newAlert()
+                                .setConfidence(Alert.RISK_INFO)
+                                .setName(
+                                        Constant.messages.getString(
+                                                "ascanalpha.forbiddenBypass.name"))
+                                .setDescription(
+                                        Constant.messages.getString(
+                                                "ascanalpha.forbiddenBypass.desc"))
+                                .setOtherInfo(TMP_URI)
+                                .setMessage(req_with_payload)
+                                .raise();
+                    }
+                }
+
+                // sending payloads in Headers.
+                for (String h : hpayloads) {
+                    String TMP_URI = SCHEMA + "://" + HOST + PATH;
```

Comment:
Better do this in the last else to avoid create the string unnecessarily.

Reply:
I created this String because We need this String multiple times and using `SCHEMA + "://" + HOST + PATH;"` every time is not good also If anyone wants to make changes in future then they can understand it easily.

Reply:
The string was being overridden in the following if statements, not being used multiple times. Looks good as is now, but could be use a `StringBuilder`.